Sources/PackageModel/Destination.swift

```diff
@@ -0,0 +1,120 @@
+import Foundation
+import Basic
+
+enum HostDestinationError: Swift.Error {
+    /// Couldn't find the Xcode installation.
+    case invalidInstallation(problem: String)
+}
+
+extension HostDestinationError: CustomStringConvertible {
+    var description: String {
+        switch self {
+        case .invalidInstallation(let problem):
+            return problem
+        }
+    }
+}
+
+enum DestinationFileError: Swift.Error {
+    /// Wrong version.
+    case destinationFileVersionMismatch(problem: String)
+}
+
+extension DestinationFileError: CustomStringConvertible {
+    var description: String {
+        switch self {
+        case .destinationFileVersionMismatch(let problem):
+            return problem
+        }
+    }
+}
+
+public struct Destination {
+    public var target: String
+
+    public var sdk: AbsolutePath
+
+    public var toolchain: AbsolutePath
+
+    public var dynamicLibraryExtension: String
+
+    public var extraCCFlags: [String]
+
+    public var extraSwiftCFlags: [String]
```

Comment:
I think its better to start with `let` unless you really need `var` because it is part of the API. If something is `let`, clients won't have a question "what happens if I change the value".  It also helps in case you need to convert the struct to a class.